### PR TITLE
Watch for post removals and push them to Blossom

### LINKED ISSUE #28: Mark posts as removed on Blossom when removed on Reddit
When a post is removed from the queue on Reddit, it should also be marked as removed on Blossom.

This is needed for both the new transcription app and the Buttercup commands to work as expected.

### PR DESCRIPTION
**Pull request by TimJentzsch**
Closes #28.

**The first commit just contains formatting fixes, you can skip it when comparing changes!**

This PR is the first step to synchronizing the post removals on Reddit with the Blossom queue.
It will look at the modlog in each update cycle and set the `removed_from_queue` flag on the corresponding submissions.

This also introduces more configuration options for the bot (via the `.env` file):
- `UPDATE_DELAY_SEC` to set the delay between each update cycle. This now defaults to 5 minutes as we now need a shorter delay to check for the removals.
- `TOR_SUBREDDIT` to configure the queue subreddit. Defaults to `TranscribersOfReddit`. This is not impacted by the `debug` option anymore!
- `DISABLE_COMPLETED_ARCHIVING`, `DISABLE_EXPIRED_ARCHIVING` and `DISABLE_POST_REMOVAL_TRACKING` to disable specific features of the archiver. All default to `False`.

### CODE REVIEW

**Review by itsthejoker — DISMISSED** [review]
just some minor formatting updates but overall this looks great, go ahead and merge when you're done

File: tor_archivist/main.py [hunk 1]
```diff
@@ -96,26 +107,64 @@ def archive_completed_posts(cfg: Config) -> None:
...
 
+def track_post_removal(cfg: Config) -> None:
+    """Process the mod log and sync post removals to Blossom."""
+    for log in cfg.tor.mod.log(action="removelink", limit=100):
+        mod = log.mod
+        tor_url = "https://reddit.com" + log.target_permalink
+        create_time = datetime.datetime.fromtimestamp(log.created_utc)
+
+        if mod.name.casefold() in ["tor_archivist", "blossom"]:
+            # Ignore our bots to avoid doing the same thing twice
```
Comment (itsthejoker) [inline tor_archivist/main.py:135]: very nice

File: tor_archivist/core/helpers.py [hunk 2]
```diff
@@ -87,12 +86,11 @@ def signal_handler(signal, frame):
...
 
     if not running:
-        logging.critical('User pressed CTRL+C twice!!! Killing!')
+        logging.critical("User pressed CTRL+C twice!!! Killing!")
         sys.exit(1)
 
     logging.info(
-        '\rUser triggered command line shutdown. Will terminate after current '
-        'loop.'
+        "\rUser triggered command line shutdown. Will terminate after current " "loop."
```
Comment (itsthejoker) [inline tor_archivist/core/helpers.py]: can we remove the extra quotes in this line

**Comment by itsthejoker** [comment]
Just ignore the version number stuff, we can do cleanup related to that afterwards as a separate thing, maybe for Shirley

File: tor_archivist/core/helpers.py (hunk 2 above)
Comment (TimJentzsch) [reply tor_archivist/core/helpers.py]: We sure can!

**Review by itsthejoker — APPROVED** [review]
This _looks_ fine -- I wish we had testing support so that we could actually verify, but if the archivist explodes then it's not the worst thing.

**Pull request merged**